<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ConnectionDialog</class>
 <widget class="QDialog" name="ConnectionDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>远程桌面连接</string>
  </property>
  <property name="windowIcon">
  <iconset resource="../../resources/resources.qrc">
   <normaloff>:/icons/connect.svg</normaloff>:/icons/connect.svg</iconset>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>10</number>
   </property>
   <property name="leftMargin">
    <number>15</number>
   </property>
   <property name="topMargin">
    <number>15</number>
   </property>
   <property name="rightMargin">
    <number>15</number>
   </property>
   <property name="bottomMargin">
    <number>15</number>
   </property>
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="connectionTab">
      <attribute name="title">
       <string>连接信息</string>
      </attribute>
      <layout class="QVBoxLayout" name="connectionLayout">
       <property name="spacing">
        <number>15</number>
       </property>
       <item>
        <widget class="QGroupBox" name="serverGroup">
         <property name="title">
          <string>服务器信息</string>
         </property>
         <layout class="QFormLayout" name="serverFormLayout">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::ExpandingFieldsGrow</enum>
          </property>
          <property name="labelAlignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="hostLabel">
            <property name="text">
             <string>主机地址:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="hostLineEdit">
            <property name="placeholderText">
             <string>输入IP地址或主机名</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="portLabel">
            <property name="text">
             <string>端口:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="portSpinBox">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>65535</number>
            </property>
            <property name="value">
             <number>5900</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="authGroup">
         <property name="title">
          <string>身份验证</string>
         </property>
         <layout class="QFormLayout" name="authFormLayout">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::ExpandingFieldsGrow</enum>
          </property>
          <property name="labelAlignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="usernameLabel">
            <property name="text">
             <string>用户名:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="usernameLineEdit">
            <property name="placeholderText">
             <string>输入用户名（可选）</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="passwordLabel">
            <property name="text">
             <string>密码:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="passwordLineEdit">
            <property name="echoMode">
             <enum>QLineEdit::Password</enum>
            </property>
            <property name="placeholderText">
             <string>输入密码（可选）</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QCheckBox" name="savePasswordCheckBox">
            <property name="text">
             <string>保存密码</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="connectionNameGroup">
         <property name="title">
          <string>连接名称</string>
         </property>
         <layout class="QFormLayout" name="nameFormLayout">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::ExpandingFieldsGrow</enum>
          </property>
          <property name="labelAlignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="nameLabel">
            <property name="text">
             <string>名称:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="nameLineEdit">
            <property name="placeholderText">
             <string>为此连接输入一个名称</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="connectionSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="optionsTab">
      <attribute name="title">
       <string>连接选项</string>
      </attribute>
      <layout class="QVBoxLayout" name="optionsLayout">
       <property name="spacing">
        <number>15</number>
       </property>
       <item>
        <widget class="QGroupBox" name="displayGroup">
         <property name="title">
          <string>显示设置</string>
         </property>
         <layout class="QFormLayout" name="displayFormLayout">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::ExpandingFieldsGrow</enum>
          </property>
          <property name="labelAlignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="colorDepthLabel">
            <property name="text">
             <string>颜色深度:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="colorDepthComboBox">
            <item>
             <property name="text">
              <string>16位 (65K 颜色)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>24位 (16M 颜色)</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>32位 (真彩色)</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QCheckBox" name="fullScreenCheckBox">
            <property name="text">
             <string>全屏模式</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QCheckBox" name="viewOnlyCheckBox">
            <property name="text">
             <string>仅查看模式（禁用输入）</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="performanceGroup">
         <property name="title">
          <string>性能设置</string>
         </property>
         <layout class="QFormLayout" name="performanceFormLayout">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::ExpandingFieldsGrow</enum>
          </property>
          <property name="labelAlignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="compressionLabel">
            <property name="text">
             <string>压缩级别:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSlider" name="compressionSlider">
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>9</number>
            </property>
            <property name="value">
             <number>6</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="tickPosition">
             <enum>QSlider::TicksBelow</enum>
            </property>
            <property name="tickInterval">
             <number>1</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="qualityLabel">
            <property name="text">
             <string>图像质量:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSlider" name="qualitySlider">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
            <property name="value">
             <number>85</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="tickPosition">
             <enum>QSlider::TicksBelow</enum>
            </property>
            <property name="tickInterval">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QCheckBox" name="adaptiveQualityCheckBox">
            <property name="text">
             <string>自适应质量</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="featuresGroup">
         <property name="title">
          <string>功能选项</string>
         </property>
         <layout class="QVBoxLayout" name="featuresVLayout">
          <item>
           <widget class="QCheckBox" name="clipboardCheckBox">
            <property name="text">
             <string>启用剪贴板同步</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="audioCheckBox">
            <property name="text">
             <string>启用音频传输</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="fileTransferCheckBox">
            <property name="text">
             <string>启用文件传输</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="cursorCheckBox">
            <property name="text">
             <string>显示远程光标</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="optionsSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="advancedTab">
      <attribute name="title">
       <string>高级设置</string>
      </attribute>
      <layout class="QVBoxLayout" name="advancedLayout">
       <property name="spacing">
        <number>15</number>
       </property>
       <item>
        <widget class="QGroupBox" name="securityGroup">
         <property name="title">
          <string>安全设置</string>
         </property>
         <layout class="QFormLayout" name="securityFormLayout">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::ExpandingFieldsGrow</enum>
          </property>
          <property name="labelAlignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
          <item row="0" column="1">
           <widget class="QCheckBox" name="encryptionCheckBox">
            <property name="text">
             <string>启用加密连接</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="encryptionTypeLabel">
            <property name="text">
             <string>加密类型:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="encryptionTypeComboBox">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <item>
             <property name="text">
              <string>AES-256</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>AES-128</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>RSA</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QCheckBox" name="verifyHostCheckBox">
            <property name="text">
             <string>验证主机身份</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="networkGroup">
         <property name="title">
          <string>网络设置</string>
         </property>
         <layout class="QFormLayout" name="networkFormLayout">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::ExpandingFieldsGrow</enum>
          </property>
          <property name="labelAlignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="timeoutLabel">
            <property name="text">
             <string>连接超时:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <layout class="QHBoxLayout" name="timeoutLayout">
            <item>
             <widget class="QSpinBox" name="timeoutSpinBox">
              <property name="suffix">
               <string> 秒</string>
              </property>
              <property name="minimum">
               <number>5</number>
              </property>
              <property name="maximum">
               <number>300</number>
              </property>
              <property name="value">
               <number>30</number>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="timeoutSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item row="1" column="1">
           <widget class="QCheckBox" name="autoReconnectCheckBox">
            <property name="text">
             <string>自动重连</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="reconnectIntervalLabel">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>重连间隔:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <layout class="QHBoxLayout" name="reconnectLayout">
            <item>
             <widget class="QSpinBox" name="reconnectIntervalSpinBox">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="suffix">
               <string> 秒</string>
              </property>
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>60</number>
              </property>
              <property name="value">
               <number>5</number>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="reconnectSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="advancedSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>

    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="buttonLayout">
     <item>
      <widget class="QPushButton" name="testConnectionButton">
       <property name="text">
        <string>测试连接</string>
       </property>
       <property name="icon">
      <iconset resource="../../resources/resources.qrc">
         <normaloff>:/icons/test.svg</normaloff>:/icons/test.svg</iconset>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="buttonSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="saveButton">
       <property name="text">
        <string>保存</string>
       </property>
       <property name="icon">
      <iconset resource="../../resources/resources.qrc">
         <normaloff>:/icons/save.svg</normaloff>:/icons/save.svg</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../../resources/resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>ConnectionDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>ConnectionDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>encryptionCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>encryptionTypeLabel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>100</x>
     <y>130</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>encryptionCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>encryptionTypeComboBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>300</x>
     <y>130</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>autoReconnectCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>reconnectIntervalLabel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>200</y>
    </hint>
    <hint type="destinationlabel">
     <x>100</x>
     <y>230</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>autoReconnectCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>reconnectIntervalSpinBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>200</y>
    </hint>
    <hint type="destinationlabel">
     <x>300</x>
     <y>230</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>